pull_target.go

```diff
@@ -0,0 +1,190 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/phrase/phraseapp-go/phraseapp"
+)
+
+type Targets []*Target
+
+func (targets Targets) ProjectIds() []string {
+	projectIds := []string{}
+	for _, target := range targets {
+		projectIds = append(projectIds, target.ProjectID)
+	}
+	return projectIds
+}
+
+type Target struct {
+	File          string
+	ProjectID     string
+	AccessToken   string
+	FileFormat    string
+	Params        *PullParams
+	RemoteLocales []*phraseapp.Locale
+}
+
+func (target *Target) CheckPreconditions() error {
+	if err := ValidPath(target.File, target.FileFormat, ""); err != nil {
+		return err
+	}
+
+	preconditions := []func(*Target) error{
+		containsStars,
+		containsDuplicatePlaceholders,
+		containsAmbiguousLocaleInformation,
+		containsInvalidTagInformation,
+	}
+
+	for _, precondition := range preconditions {
+		if err := precondition(target); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func containsStars(target *Target) error {
+	if strings.Count(target.File, "*") > 0 {
+		return fmt.Errorf("File pattern for 'pull' cannot include any 'stars' *. Please specify direct and valid paths with file name!\n %s#targets", docsConfigUrl)
+	}
+	return nil
+}
+
+func containsDuplicatePlaceholders(target *Target) error {
+	duplicatedPlaceholders := []string{}
+	for _, name := range []string{"<locale_name>", "<locale_code>", "<tag>"} {
+		if strings.Count(target.File, name) > 1 {
+			duplicatedPlaceholders = append(duplicatedPlaceholders, name)
+		}
+	}
+
+	if len(duplicatedPlaceholders) > 0 {
+		dups := strings.Join(duplicatedPlaceholders, ", ")
+		return fmt.Errorf(fmt.Sprintf("%s can only occur once in a file pattern!", dups))
```

Comment:
no need for having an extra `fmt.Sprintf` here, this is what `fmt.Errorf` would already do for us